fabric-entity-events-v1/src/main/java/net/fabricmc/fabric/api/entity/event/v1/EntityElytraEvents.java

```diff
@@ -0,0 +1,116 @@
+
+/**
+ * Events related to elytra flight for living entities. Elytra flight is also knows as "fall flying".
+ */
+public class EntityElytraEvents {
+	/**
+	 * An event that allows listeners to grant elytra flight to living entities.
+	 * Will be called when players try to start elytra flight by pressing space in mid-air, and every tick for all living entities to check if elytra flight is still allowed.
+	 *
+	 * <p>Items that wish to grant elytra flight when worn in the chest equipment slot can simply implement {@link FabricElytraItem} instead of registering a listener.
+	 */
+	public static final Event<Grant> GRANT = EventFactory.createArrayBacked(Grant.class, listeners -> (livingEntity, damageElytra) -> {
+		for (Grant listener : listeners) {
+			if (listener.grantElytraFlight(livingEntity, damageElytra)) {
+				return true;
+			}
+		}
+
+		return false;
+	});
+
+	static {
+		GRANT.register((livingEntity, damageElytra) -> {
+			ItemStack chestStack = livingEntity.getEquippedStack(EquipmentSlot.CHEST);
+
+			if (chestStack.getItem() instanceof FabricElytraItem fabricElytraItem) {
+				return fabricElytraItem.grantElytraFlight(livingEntity, chestStack, damageElytra);
+			}
+
+			return false;
+		});
+	}
+
+	@FunctionalInterface
+	public interface Grant {
+		/**
+		 * Check if elytra flight should be granted to an entity.
+		 *
+		 * @param livingEntity the entity
+		 * @param tickElytra   false if this is just to check if elytra flight is allowed, true if the elytra should be ticked, i.e. damaged.
+		 * @return true to grant elytra flight, and cancel subsequent handlers
+		 */
+		boolean grantElytraFlight(LivingEntity livingEntity, boolean tickElytra);
+	}
+
+	/**
+	 * An interface that can be implemented on an item to enable elytra flight when it is worn in the {@link EquipmentSlot#CHEST} slot.
+	 *
+	 * <p>To disable cape rendering when this item is worn (like the vanilla elytra item), have a look at {@code LivingEntityFeatureRenderEvents}.
+	 */
+	public interface FabricElytraItem {
+		/**
+		 * Check if the item grants elytra flight to an entity.
+		 *
+		 * @param livingEntity the entity
+		 * @param chestStack   the stack currently worn in the chest slot, will always be of this item
+		 * @param tickElytra   true to tick the elytra, false to only perform the check; vanilla-like elytras can use {@link #doVanillaElytraTick} to handle ticking
+		 * @return true to grant elytra flight
+		 */
+		default boolean grantElytraFlight(LivingEntity livingEntity, ItemStack chestStack, boolean tickElytra) {
+			if (ElytraItem.isUsable(chestStack)) {
+				if (tickElytra) {
+					doVanillaElytraTick(livingEntity, chestStack);
+				}
+
+				return true;
+			}
+
+			return false;
+		}
+
+		/**
+		 * A helper to perform the default vanilla elytra tick logic: damage the elytra every 20 ticks, and send a game event every 10 ticks.
+		 */
+		default void doVanillaElytraTick(LivingEntity livingEntity, ItemStack chestStack) {
```

Comment:
I don't think it's worth copying the vanilla code, as it will make it difficult to update the module to the new version of the game.

This API will break my mod, which completely overrides the elytra damage logic, so this is very important to me.
https://gist.github.com/maityyy/05a7656e363cd347c6e583646a32a0f9

I think if someone needs this functionality, they can use the mixin without any problems, which will not break my mod.

Reply:
There's a lot of conflicts in how mixins are written and how tightly they're written together. My mod (Winged) had quite a lot of problems regarding mod compatibility between other mods, solely because of how many mixins ends up being mashed together. I tried to fix this with FFL and ultimately failed, and I ultimately prefer that the API allows for the least amount of conflicts between mods trying to implement Elytras and elytra-related functionality.

Reply:
Also this code don't interfere with the vanilla elytra funcionality, and would only break cases of modded elytras + your mod's feature, which I think is fair enough. Ask modders if they want their custom wings to have compatibility with your mod, and just accept the case that not everyone will want such compatibility.